hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -142,16 +146,22 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
     String newMetadataLocation = writeNewMetadata(metadata, currentVersion() + 1);
 
     boolean threw = true;
+    boolean updateHiveTable = false;
     Optional<Long> lockId = Optional.empty();
     try {
       lockId = Optional.of(acquireLock());
       // TODO add lock heart beating for cases where default lock timeout is too low.
       Table tbl;
-      if (base != null) {
-        LOG.debug("Committing existing table: {}", fullName);
+      try {
         tbl = metaClients.run(client -> client.getTable(database, tableName));
+        if (base == null && tbl.getParameters().get(TABLE_CREATION_FROM_HIVE) == null) {
+          throw new AlreadyExistsException("Table already exists: %s.%s", database, tableName);
+        }
+        updateHiveTable = true;
+        LOG.debug("Committing existing table: {}", fullName);
         tbl.setSd(storageDescriptor(metadata)); // set to pickup any schema changes
-      } else {
+      } catch (NoSuchObjectException nte) {
+        LOG.trace("Table not found {}", fullName, nte);
```

Comment:
Can we remove this? If the debug log shows that this path was taken, then it is redundant.

Reply:
Can we suppress the warning instead?
```
> Task :iceberg-hive-metastore:compileJava
/Users/petervary/dev/upstream/iceberg/hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java:163: error: [CatchBlockLogException] Catch block contains log statements but thrown exception is never logged.
      } catch (NoSuchObjectException nte) {
        ^
    (see https://github.com/palantir/gradle-baseline#baseline-error-prone-checks)
  Did you mean 'LOG.debug("Committing new table: {}", fullName, nte);'?
```
Or we should just log the exception on debug level?
I wanted to dig the exception as deep as possible, as it is normal to have it 😄

Reply:
It's fine to suppress the exception. I just would normally not bother logging it.

Reply:
Actually, I think you're right. Let's just log it at debug. I see what you're saying now.